src/js/components/utils/deleteModal.tsx

```diff
@@ -0,0 +1,63 @@
+import Button from '@salesforce/design-system-react/components/button';
+import Modal from '@salesforce/design-system-react/components/modal';
+import i18n from 'i18next';
+import React from 'react';
+import { Trans } from 'react-i18next';
+
+import { Project } from '@/store/projects/reducer';
+import { Task } from '@/store/tasks/reducer';
+import { OBJECT_TYPES } from '@/utils/constants';
+
+const DeleteModal = ({
+  model,
+  instanceType,
+  isOpen,
+  handleClose,
+  handleDelete,
+}: {
+  model: Project | Task;
+  instanceType: 'project';
+  isOpen: boolean;
+  handleClose: () => void;
+  handleDelete: () => void;
+}) => {
+  const handleConfirmDelete = () => {
+    handleClose();
+    handleDelete();
```

Comment:
Maybe there should be some loading states here, but I think I'm calling close both here and in the detail page.  This would need to change anyway if we need to call `deleteObject` action from here.